### DISCUSSION TITLE
Sunbird video player progress and End event summary changes

### DISCUSSION THREAD

**Discussion by HarishGangula** [Sunbird-Knowlg/Community#91] [General Discussions]
<p>Currently, The summary data from the telemetry end event contains</p><figure class="table" style="height:1px;width:760px;">

<table style="background-color:rgb(255, 255, 255);border:1px solid var(--ds-border, #C1C7D0);">
    <tbody>
        <tr>
            <th style="background-color:var(--ds-background-neutral, #F4F5F7);padding:0px;vertical-align:top;" colspan="1" rowspan="1">
                <p>
                    <strong>Property</strong>
                </p>
                <p>
                    &nbsp;
                </p>
            </th>
            <td style="background-color:var(--ds-background-neutral, #F4F5F7);padding:0px;vertical-align:top;" colspan="1" rowspan="1">
                <p>
                    <strong>Type</strong>
                </p>
                <p>
                    &nbsp;
                </p>
            </td>
            <td style="background-color:var(--ds-background-neutral, #F4F5F7);padding:0px;vertical-align:top;" colspan="1" rowspan="1">
                <p>
                    <strong>Description</strong>
                </p>
                <p>
                    &nbsp;
                </p>
            </td>
        </tr>
        <tr>
            <th style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                <strong>progress</strong>
            </th>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Number
            </td>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                This is the actual progress calculated by the player
            </td>
        </tr>
        <tr>
            <th style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                <strong>endpageseen</strong>
            </th>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Boolean
            </td>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                “true“ if the user visited player end page
            </td>
        </tr>
        <tr>
            <th style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                <strong>visitedcontentend</strong>
            </th>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Boolean
            </td>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                “true“ if the user visited the content end page
            </td>
        </tr>
        <tr>
            <th style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                <strong>totallength</strong>
            </th>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Number
            </td>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                The total length of content
            </td>
        </tr>
        <tr>
            <th style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                <strong>visitedlength</strong>
            </th>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Number
            </td>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                The max-content length covered by the user
            </td>
        </tr>
        <tr>
            <th style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                <strong>totalintrections</strong>
            </th>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Number
            </td>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Total number of interactions triggered by the player while consuming content
            </td>
        </tr>
        <tr>
            <th style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                <strong>duration</strong>
            </th>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Number
            </td>
            <td style="padding:8px;vertical-align:top;" colspan="1" rowspan="1">
                Total time spent on the content player to play the content(includes pause time as well)
            </td>
        </tr>
    </tbody>
</table>

</figure><p>&nbsp;</p><p>When calculating progress if the visistedLength is taken then with the introduction of the change in play speed player's and with replays it affect so there is need to calculate the progress based on the unique time spent by the user irrespective of the player speed at which user consumed the content or any number of the pause or seeking happened back and forth, The new key in the summary event will be <strong>uniqueVisitedLength </strong>which will used to calculate the progress.<br><br>&nbsp;</p><p><br>&nbsp;</p>

cc: @pallakartheekreddy @Krishnaj20 @vrayulu @vinukumar-vs @rajeevsathish @swayangjit

**Comment by Krishnaj20** [comment]
@HarishGangula - unique time spent will help us derive the actual progress of user when viewing a video under different scenarios - user playing at 2x speed, user dragging the seeker and forwarding or replaying what they've seen etc. When is this feature planned to go live - in which Knowlg release? 

@rajeevsathish @swayangjit @vinukumar-vs - This will have an impact on the consumption side as well in how the current progress calculations are done. Please review and share your feedback/questions.

**Reply by HarishGangula** [reply]
@Krishnaj20  we are planning to complete this as part of Knowlg 5.4.0 release

**Reply by vinukumar-vs** [reply]
@HarishGangula I am still not clear with "uniqueVisitedLength". 
What will the value of it when the user played with multiple pauses for some minutes?
What will be the value of it when the user drags the video?
What will be the value of it when the user completed watching & replay it again. But he is exiting from middle of video?

how this unique timespend will help to address the progress issue?
Is it same as maxVisitedLength of the video?

@Krishnaj20 @rajeevsathish @swayangjit  @pallakartheekreddy @Krishnaj20 @vrayulu

**Reply by HarishGangula** [reply]
**uniqueVisitedLength:**  it is unique duration of the video played by the user   irrespective of speed or pause and plays or seeks

What will the value of it when the user played with multiple pauses for some minutes?

`Even with multiple pause it will calculate the unique time spent by the user in that video session`
	
 What will be the value of it when the user drags the video?

	`For 2 min video, If user starts video at 0 and watched till 20 sec then he dragged video 1:50 then completed video that is till 2 min then uniqueVisitedLength = (20 + 10)`
	
 What will be the value of it when the user completed watching & replay it again. But he is exiting from middle of video? 	
 
	`Continues to watch is considered as one session and replay is another session uniqueVisitedLength value will be different for each session `
	
how this unique timespend will help to address the progress issue? 

`With uniqueVisitedLength now we will know exactly how much duration he covered irrespective of the user dragging back and forth of the video here watching same video bit again won’t be considers which helps in the calculation of the progress  `

Is it same as maxVisitedLength of the video?  

`No, maxVisitedLength = total duration seen by user including the repeated duration seen`

**Reply by HarishGangula** [reply]
to keep all the keys in summary consistency we will use **uniquevisitedlength** instead of **uniqueVisitedLength**
